lib/install/config/loaders/core/assets.js

```diff
@@ -1,11 +1,21 @@
-const { env, publicPath } = require('../configuration.js')
+const Url = require('url-parse')
+const { env, paths, publicPath } = require('../configuration.js')
+
+// Compute path for internal pack assets if ASSET_HOST is supplied
+const computeAssetPath = (host) => {
+  if (!host) { return `/${paths.entry}/` }
+  const { slashes, href } = new Url(host)
+  if (slashes) { return `${href}/${paths.entry}/` }
+  if (href) { return `//${href}/${paths.entry}/` }
+  return `/${paths.entry}/`
+}
 
 module.exports = {
   test: /\.(jpg|jpeg|png|gif|svg|eot|ttf|woff|woff2)$/i,
   use: [{
     loader: 'file-loader',
     options: {
-      publicPath,
+      publicPath: env.NODE_ENV === 'production' ? computeAssetPath(env.ASSET_HOST) : publicPath,
       name: env.NODE_ENV === 'production' ? '[name]-[hash].[ext]' : '[name].[ext]'
```

Comment:
Since both of these options differ on the same conditional, we should extract the conditional to a higher level. Maybe:

```javascript
cont assetOptions = (env) => {
  if (env.NODE_ENV === 'production') {
    return { publicPath: computeAssetPath(env.ASSET_HOST), name: '[name]-[hash].[ext]' }
  } else {
    return { publicPath: publicPath, name: '[name].[ext]' }
  }
}

options: assetOptions(env)
```